### ISSUE TITLE
Comment() currently does nothing

### ISSUE THREAD

**Issue by moqmar** [facebook/ent#944]
The documentation says "Comment sets the comment of the field.", but [it doesn't seem to set anything](https://github.com/facebook/ent/blob/master/schema/field/numeric.go#L185) currently.

What I would have expected is that it changes the actual comment of the field, i.e. modify the not-very-helpful default comment "Title holds the value of the "title" field." - this would for example make the `Comment()` method extremely helpful when using something like [swag](https://github.com/swaggo/swag) to document an API, as that comment will be parsed when generating the model documentation.

**Comment by a8m** [comment]
Hey @moqmar,
I agree, we should change this.

Do you want to contribute this change? If so, I'll add code-pointers with some guidance tomorrow.

**Comment by moqmar** [comment]
Sure, I can open a pull request for that! I guess it's more or less just adding a `Comment string` field to the [Descriptor](https://github.com/facebook/ent/blob/master/schema/field/field.go#L885) type and using it in the [ent.tmpl](https://github.com/facebook/ent/blob/master/entc/gen/template/ent.tmpl) template if it's non-empty, or am I missing anything here?

Things I can think of that I have to take into account:
- Newlines should be handled correctly
- The comment should start with the field name according to the GoDoc rules - should this automatically be prefixed? Possible solutions I can think of:
  - `field.String("title").Comment("Title is your superduperawesome headline")` - what to do if it doesn't start with the struct name?
  - `field.String("title").Comment("is your superduperawesome headline")` - isn't that intuitive IMO.
  - require the first word to be the struct name, otherwise prefix the comment with the struct name and a dash (e.g. `// Title - The article's title`) - probably makes the most sense?

**Comment by a8m** [comment]
> Things I can think of that I have to take into account:
> 
> * Newlines should be handled correctly
> * The comment should start with the field name according to the GoDoc rules - should this automatically be prefixed? Possible solutions I can think of:
>   
>   * `field.String("title").Comment("Title is your superduperawesome headline")` - what to do if it doesn't start with the struct name?
>   * `field.String("title").Comment("is your superduperawesome headline")` - isn't that intuitive IMO.
>   * require the first word to be the struct name, otherwise prefix the comment with the struct name and a dash (e.g. `// Title - The article's title`) - probably makes the most sense?

Yeah, we should handle newlines, but this should be simple. Regarding other issues, it's a field comment and I think we should keep it simple, and not come up with a super smart solution (i.e. paste the comment as-is prefixed with `// `). 

Code pointers:
1\. Like you mentioned, add the `Comment` value to the `field.Descriptor` - can be in a separate PR.
2\. Add the comment to the `Field` object in the loaded-schema and run `go generate` on package `entc/load` (I prefer a separate PR as well). 
https://github.com/facebook/ent/blob/6b2673f82dcfe3f1d72f889e12b468d2741f2428/entc/load/schema.go#L39-L58
3\. In `ent.tmpl` template, paste the comment (you need to expose it using a method on the field) or fallback to default - you can use the `split` function in the template to handle multiline comments.

**Comment by a8m** [comment]
Solved with #1205.

Thanks @kerbelp for your contribution.

**Comment by Colstuwjx** [comment]
Could us reopen this issue? I've tried to generate the sql with the following statements:

```
client.Schema.WriteTo(
			context.Background(),
			f,
			migrate.WithForeignKeys(false),
		)
```

and when I add comment for a field, e.g. `field.Int64("id").Comment("primary key")`, it still didn't work, I can't see any comments in the generated sql.
